<launch>
  <node pkg="image_view2" name="attention_viewer" type="image_view2">
    <remap from="image" to="/multisense/left/image_rect_color" />
    <rosparam>
      interaction_mode: line
    </rosparam>
  </node>
  <node pkg="jsk_pcl_ros" type="tf_transform_cloud" name="laser_cloud_from_camera_frame">
    <remap from="~input" to="/full_cloud2" />
    <rosparam>
      target_frame_id: multisense/left_camera_optical_frame
    </rosparam>
  </node>
  <node pkg="jsk_pcl_ros" type="hinted_stick_finder" name="hinted_stick_finder" output="screen">
    <remap from="~input" to="laser_cloud_from_camera_frame/output" />
    <remap from="~input/hint/line" to="/multisense/left/image_rect_color/line" />
    <remap from="~input/camera_info" to="/multisense/left/camera_info" />
  </node>

  <node pkg="nodelet" type="nodelet" name="candidate_points"
        args="standalone pcl/ExtractIndices" output="screen">
    <remap from="~input" to="laser_cloud_from_camera_frame/output" />
    <remap from="~indices" to="hinted_stick_finder/debug/line_filtered_indices" />
    <rosparam>
      use_indices: true
      approximate_sync: true
    </rosparam>
  </node>
  
</launch>
